core/service_test.go

```diff
@@ -30,73 +31,105 @@ import (
 	"github.com/ChainSafe/gossamer/keystore"
 	"github.com/ChainSafe/gossamer/network"
 	"github.com/ChainSafe/gossamer/runtime"
+	"github.com/ChainSafe/gossamer/state"
 	"github.com/ChainSafe/gossamer/tests"
 	"github.com/ChainSafe/gossamer/trie"
 )
 
-var TestMessageTimeout = 2 * time.Second
+var TestMessageTimeout = 5 * time.Second
 
-func TestStartService(t *testing.T) {
-	rt := runtime.NewTestRuntime(t, tests.POLKADOT_RUNTIME)
+func newTestService(t *testing.T, cfg *Config) (*Service, *state.Service) {
+	if cfg == nil {
+		rt := runtime.NewTestRuntime(t, tests.POLKADOT_RUNTIME)
 
-	cfg := &Config{
-		Runtime:         rt,
-		Keystore:        keystore.NewKeystore(),
-		MsgRec:          make(chan network.Message),
-		MsgSend:         make(chan network.Message),
-		IsBabeAuthority: false,
+		cfg = &Config{
+			Runtime:         rt,
+			IsBabeAuthority: false,
+		}
 	}
 
-	s, err := NewService(cfg)
+	if cfg.Keystore == nil {
+		cfg.Keystore = keystore.NewKeystore()
+	}
+
+	if cfg.MsgRec == nil {
+		cfg.MsgRec = make(chan network.Message)
+	}
+
+	if cfg.MsgSend == nil {
+		cfg.MsgSend = make(chan network.Message)
+	}
+
+	dataDir, err := ioutil.TempDir("", "./test_data")
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	err = s.Start()
+	if cfg.NewBlocks == nil {
+		cfg.NewBlocks = make(chan types.Block)
+	}
+
+	dbSrv := state.NewService(dataDir)
+	err = dbSrv.Initialize(&types.Header{
+		Number:    big.NewInt(0),
+		StateRoot: trie.EmptyHash,
+	}, trie.NewEmptyTrie(nil))
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	s.Stop()
-}
+	err = dbSrv.Start()
+	if err != nil {
+		t.Fatal(err)
+	}
 
-func TestValidateBlock(t *testing.T) {
-	rt := runtime.NewTestRuntime(t, tests.POLKADOT_RUNTIME)
+	// defer func() {
+	// 	err = dbSrv.Stop()
+	// 	if err != nil {
+	// 		t.Fatal(err)
+	// 	}
+	// }()
```

Comment:
Why is this commented out?

Reply:
shouldn't be here, this function shouldn't stop the db service, but the tests that use it should